src/workerd/io/trace.c++

```diff
@@ -1536,11 +1535,39 @@ void SpanBuilder::setOperationName(kj::ConstString operationName) {
   }
 }
 
-void SpanBuilder::setTag(kj::ConstString key, TagValue value) {
+void SpanBuilder::setTag(kj::ConstString key, TagInitValue value) {
   KJ_IF_SOME(s, span) {
+    // We allow passing a LiteralStringConst or StringPtr so that we don't have to allocate memory
+    // if we're not being observed.
```

Comment:
This looks like it would allocate whenever a kj::StringPtr was passed, ~~or any ConstString that itself contains an allocated string~~.  I'm not sure how it would avoid allocations when not being observed, unless the caller has logic to pass different values in those cases?

Reply:
Note that span is only non-null if we are being observed – in the SpanBuilder constructor, we already avoid allocating it if we are not observed, so setTag is a no-op in that case.